File: 2024/01.test.jl
include("01.jl")
using Test

example_input = """
3   4
4   3
2   5
1   3
3   9
3   3"""

@testset "prepare_input" begin
    @test prepare_input(example_input) == [3 4 2 1 3 3; 4 3 5 3 9 3]
end

@testset "Example Input" begin
    input = prepare_input(example_input)
    @test part1(input) == 11
    @test part2(input) == 31
end

@testset "Real input" begin
    input = prepare_input(get_input(day=1, year=2024))
    @test part1(input) == 2742123
    @test part2(input) == 21328497
end

File: 2025/test/01_test.rb
# frozen_string_literal: true

require 'minitest/autorun'
require_relative '../01'

# :nodoc:
class TestDay01 < Minitest::Test
  include Day01

  def initialize(...)
    super(...)
    @example_input = <<~INPUT
      L68
      L30
      R48
      L5
      R60
      L55
      L1
      L99
      R14
      L82
    INPUT
  end

  def test_prepare_input
    assert_equal(
      [-68, -30, +48, -5, +60, -55, -1, -99, +14, -82],
      prepare_input(@example_input)
    )
  end

  def test_example_input
    input = prepare_input(@example_input)
    assert_equal(3, part1(input))
  end

  def test_real_input
    input = prepare_input(get_input(1, 2025))
    assert_equal(1180, part1(input))
  end
end
